driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionCorpusTest.java

```diff
@@ -251,31 +248,34 @@ public void testCorpus() throws IOException, URISyntaxException {
     }
 
     // Check that all values of document that are deterministic with the same kms + type match value
-    private static void deterministicCheck(final String matchKms, final String matchType, final BsonDocument document) {
-        for (String field : document.keySet()) {
+    private static void deterministicCheck(final BsonValue actualValue, final String actualKms, final String actualType,
+                                           final BsonDocument expectedDocument) {
+        for (String field : expectedDocument.keySet()) {
             if (field.equals("_id") || field.equals("altname_aws") || field.equals("altname_local")) {
                 continue;
             }
 
-            BsonDocument subDocument = document.getDocument(field);
+            BsonDocument subDocument = expectedDocument.getDocument(field);
 
             String kms = subDocument.getString("kms").getValue();
             String type = subDocument.getString("type").getValue();
-            String algo = subDocument.getString("algo").getValue();
-            BsonValue value = subDocument.get("value");
+            String algorithm = subDocument.getString("algo").getValue();
+            BsonValue expectedValue = subDocument.get("value");
 
-            if (kms.equals(matchKms) && type.equals(matchType) && algo.equals("det")) {
-                assertEquals(value, value);
+            if (kms.equals(actualKms) && type.equals(actualType) && algorithm.equals("det")) {
+                assertEquals(expectedValue, actualValue);
```

Comment:
This was just totally wrong.  Fixed to compare expected vs actual

Reply:
I've updated the randomCheck in https://github.com/mongodb/specifications/pull/604, I think we can use the same function for `deterministicCheck` and `randomCheck` and switch on this assert.

Reply:
Not sure if the spec test instructions are exactly right: it says to match on kms and type, but you also have to match on algorithm

Reply:
Ah true, I'll fix that.